src/MVVM/ViewModels/Navigation/ViewModelBaseHelper.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System.Linq;
+using System.Reflection;
+
+using CodeMonkeys.Core.Interfaces.MVVM;
+using CodeMonkeys.Core.Interfaces.Navigation;
+
+namespace CodeMonkeys.MVVM.ViewModels.Navigation
+{
+    internal static class ViewModelBaseHelper
```

Comment:
Don't know but does it make sense to cache some stuff here? Or is it not possible? If yes, why not?